src/sagemaker/workflow/step_collections.py

```diff
@@ -243,6 +244,9 @@ def __init__(
         self.steps = steps
 
 
+RegisterModel = deprecated_class(RegisterModel, "RegisterModel")
```

Comment:
Let's add a warning message here directing user to ModelStep

Reply:
Sure, added in the new commit